docs/jwks.md

```diff
@@ -0,0 +1,48 @@
+# JWKS
+
+We expose a JWKS (JSON Web Key Set) endpoint at `/oauth/jwks`. This allows
+consumers of JWTs issued by us to validate the signature, per OIDC standards.
+This was set up to suport LTI 1.3 features, however it can be used by other
+features that need to issue JWTs. Below is an overview of generating/rotating
+JWKs, both the public and private portion of the keys.
+
+## Generating a new JWK
+
+Occasionally it may be necessary to rotate a JWK, i.e. generate a new JWK and
+private key. The JWKS endpoint returns an array of public keys (JWK), which
+allows us to starting using a new JWK, while still allowing users to validate
+JWTs issued with the old JWK. To generate a new JWK, refer to the following
+steps:
+
+1. Run `bin/generate-jwks`
+1. This will add the new JWK to the JWKS in `dashboard/public/oauth/jwks.json`
+1. Use the private key data printed in your terminal to update the secret using
+   the [secrets process](../config/secrets.md).
```

Comment:
Should there be another step here around doing a PR to accomplish the rotation or is that too obvious? Also I'm assuming the script is intended to be run locally right? Because we wouldn't want to write the new JSON directly onto a prod server. Might be worth adding that warning.

Also, if a secret changes inside Secrets Manager, would a re-deploy be required for the machines to pull down the new value? I'm assuming so, which makes things a little easier, but we should probably specify here that the deployment for step 2 (new public jwks) should only be done after the new PK is added to secrets manager in step 3. Otherwise, we could end up in a state where new JWTs are being issued but consumers don't yet have access to the new JWKS with the new KID.

Reply:
Good call, yeah this needs more documentation. I'll address the questions in order:

* Yep, when doing a JWK rotation, we should remove the old JWK _after_ the TTL of the JWT has expired. So if we are issuing JWTs with an expiration of 1 hour, we shouldn't remove the old JWK until that time has passed. There is no harm in leaving a deprecated JWK around longer, as all new JWTs will reference the new JWK via the kid.
* Yes, a secret changing in Secrets Manager requires a re-deploy, as these secrets are loaded during startup. And yes, a bit of coordination is required. I'll add that to the docs